From c8d64259cb1a8dda555b8f64464b0937ad5a2b92 Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Mon, 7 Jul 2025 15:46:39 +0200
Subject: [PATCH 2/2] Reorder packaging options for singleton packaging

Packaging options related to singleton packaging must come after other
packaging options.

Forwarded: https://github.com/mizdebsk/dola/pull/22
---
 .../io/kojan/dola/imperator/Imperator.java    | 38 +++++++++----------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/dola-dbs/src/main/java/io/kojan/dola/imperator/Imperator.java b/dola-dbs/src/main/java/io/kojan/dola/imperator/Imperator.java
index 0b3a784..99b053f 100644
--- a/dola-dbs/src/main/java/io/kojan/dola/imperator/Imperator.java
+++ b/dola-dbs/src/main/java/io/kojan/dola/imperator/Imperator.java
@@ -105,25 +105,6 @@ public class Imperator {
             conf.getBuildSettings().setSkipTests(true);
         }
 
-        if (ctx.isSingletonPackaging()) {
-            PackagingRule rule1 = new PackagingRule();
-            org.fedoraproject.xmvn.config.Artifact glob1 =
-                    new org.fedoraproject.xmvn.config.Artifact();
-            glob1.setClassifier("*?");
-            rule1.setArtifactGlob(glob1);
-            rule1.setTargetPackage("__noinstall");
-            rule1.setOptional(true);
-            conf.addArtifactManagement(rule1);
-
-            PackagingRule rule2 = new PackagingRule();
-            org.fedoraproject.xmvn.config.Artifact glob2 =
-                    new org.fedoraproject.xmvn.config.Artifact();
-            glob2.setArtifactId("{*}");
-            rule2.setArtifactGlob(glob2);
-            rule2.setTargetPackage("@1");
-            conf.addArtifactManagement(rule2);
-        }
-
         for (PackagingOption po : ctx.getPackagingOptions()) {
             PackagingRule rule = new PackagingRule();
             org.fedoraproject.xmvn.config.Artifact glob =
@@ -149,6 +130,25 @@ public class Imperator {
             conf.addArtifactManagement(rule);
         }
 
+        if (ctx.isSingletonPackaging()) {
+            PackagingRule rule1 = new PackagingRule();
+            org.fedoraproject.xmvn.config.Artifact glob1 =
+                    new org.fedoraproject.xmvn.config.Artifact();
+            glob1.setClassifier("*?");
+            rule1.setArtifactGlob(glob1);
+            rule1.setTargetPackage("__noinstall");
+            rule1.setOptional(true);
+            conf.addArtifactManagement(rule1);
+
+            PackagingRule rule2 = new PackagingRule();
+            org.fedoraproject.xmvn.config.Artifact glob2 =
+                    new org.fedoraproject.xmvn.config.Artifact();
+            glob2.setArtifactId("{*}");
+            rule2.setArtifactGlob(glob2);
+            rule2.setTargetPackage("@1");
+            conf.addArtifactManagement(rule2);
+        }
+
         if (conf.getBuildSettings() != null || !conf.getArtifactManagement().isEmpty()) {
             lines.add("# Write out XMvn configuration before calling XMvn for the first time.");
             Path confPath = Path.of(".xmvn/config.d/dola.xml");
-- 
2.50.0

